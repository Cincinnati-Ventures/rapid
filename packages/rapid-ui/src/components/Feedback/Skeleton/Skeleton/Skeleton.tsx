import React from 'react';
import { RapidStyles } from '../../../../utils';
import classNames from '../../../../conditional';
import { twMerge } from 'tailwind-merge';
import type { Speed } from '..';

const RAPID_CLASSNAME = 'rapid-skeleton';

// Note: we do not want the user to be able to pass in children to the Skeleton component
interface SkeletonProps
	extends React.HTMLAttributes<Omit<HTMLDivElement, 'children'>> {
	styles?: string;
	isLoading?: boolean;
	speed?: Speed;
}

const Skeleton = React.forwardRef<HTMLDivElement, SkeletonProps>(
	({ styles, isLoading = true, speed = 'skeleton-pulse', ...rest }, ref) => {
		const defaultStyles = 'h-6 w-40 rounded-md';
		return (
			<div
				{...rest}
				ref={ref}
				className={twMerge(
					RapidStyles(
						styles || rest.className,
						defaultStyles,
						RAPID_CLASSNAME,
					),
					// The pulse className is merged with all the others so that it cannot get overridden by the user
					classNames({
						condition: isLoading,
						classNames: speed || 'skeleton-pulse',
					}),
				)}
			/>
		);
	},
);

Skeleton.displayName = 'Skeleton';

export default Skeleton;
