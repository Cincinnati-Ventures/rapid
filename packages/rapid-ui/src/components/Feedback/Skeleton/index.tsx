export { default as Skeleton } from './Skeleton/Skeleton';
export { default as SkeletonText } from './SkeletonText/SkeletonText';

export type Speed =
	| 'skeleton-pulse'
	| 'skeleton-pulse-medium'
	| 'skeleton-pulse-fast';
