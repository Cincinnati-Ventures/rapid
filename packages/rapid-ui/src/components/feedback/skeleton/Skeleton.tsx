import React from 'react';
import { RapidStyles } from '../../../utils';
import classNames from '../../../conditional';
import { twMerge } from 'tailwind-merge';

const RAPID_CLASSNAME = 'rapid-skeleton';

// Note: we do not want the user to be able to pass in children to the Skeleton component
interface SkeletonProps extends React.HTMLAttributes<Omit<HTMLDivElement, "children">> {
	styles?: string;
	isLoading?: boolean;
}

const Skeleton = React.forwardRef<HTMLDivElement, SkeletonProps>(
	({ styles, isLoading = true, ...rest }, ref) => {
		const defaultStyles = 'h-6 w-40 rounded-md';
		return (
			<div
				{...rest}
				ref={ref}
				className={twMerge(
					RapidStyles(
						styles || rest.className,
						defaultStyles,
						RAPID_CLASSNAME,
					),
					classNames({
						condition: isLoading,
						classNames: 'skeleton-pulse',
					}),
				)}
			/>
		);
	},
);

Skeleton.displayName = "Skeleton";

export default Skeleton;
