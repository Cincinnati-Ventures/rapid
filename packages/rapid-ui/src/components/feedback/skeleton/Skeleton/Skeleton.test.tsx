import React from 'react';
import { render } from '@testing-library/react';
import { expect, it, describe } from 'vitest';
import Skeleton from './Skeleton';

describe('Skeleton', () => {
	it('should render correctly', () => {
		const { container } = render(<Skeleton />);
		expect(container).toMatchSnapshot();
	});

	it('should render correctly with props', () => {
		const { container } = render(
			<Skeleton
				styles='custom-class'
				isLoading
				speed='skeleton-pulse-fast'
			/>,
		);
		expect(container).toMatchSnapshot();
	});

	it('should render correctly with props', () => {
		const { container } = render(
			<Skeleton isLoading={false} speed='skeleton-pulse-medium' />,
		);
		expect(container).toMatchSnapshot();
	});
});
