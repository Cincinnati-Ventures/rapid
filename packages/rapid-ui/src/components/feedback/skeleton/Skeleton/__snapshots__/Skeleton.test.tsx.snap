// Vitest Snapshot v1

exports[`Skeleton > should render correctly 1`] = `
<div>
  <div
    class="h-6 w-full rounded-md rapid-skeleton skeleton-pulse"
  />
</div>
`;

exports[`Skeleton > should render correctly with props 1`] = `
<div>
  <div
    class="h-6 w-full rounded-md custom-class rapid-skeleton skeleton-pulse-fast"
  />
</div>
`;

exports[`Skeleton > should render correctly with props 2`] = `
<div>
  <div
    class="h-6 w-full rounded-md rapid-skeleton"
  />
</div>
`;
